stanford_media.post_update.php

```diff
@@ -0,0 +1,355 @@
+        'label' => 'hidden',
+      ]);
+      $view_display->save();
+    }
+  }
+
+  drupal_flush_all_caches();
+
+  $source_config = [
+    'source_field' => 'field_media_oembed_video',
+    'providers' => ['YouTube', 'Vimeo'],
+    'thumbnails_directory' => 'public://oembed_thumbnails',
+  ];
+  /** @var \Drupal\media\MediaTypeInterface $media_type */
+  $media_type = MediaType::load('video');
+  $media_type->set('source_configuration', $source_config);
+  $media_type->set('source', 'oembed:video');
+  $media_type->calculateDependencies();
+  $media_type->save();
+
+}
+
+/**
+ * Update text formats by switching the media buttons.
+ */
+function stanford_media_post_update_8201() {
+
+  /** @var \Drupal\editor\EditorInterface $editor */
+  foreach (Editor::loadMultiple() as $editor) {
+
+    $settings = $editor->getSettings();
+    foreach ($settings['toolbar']['rows'] as &$row_items) {
+      foreach ($row_items as &$row_group) {
+        if ($media_browser_position = array_search('media_browser', $row_group['items'])) {
+          $row_group['items'][$media_browser_position] = 'DrupalMediaLibrary';
+          $editor->setSettings($settings);
+          $editor->save();
+        }
+      }
+    }
+  }
+
+  $display_modes = ['full'];
+  $media_settings = \Drupal::configFactory()
+    ->getEditable('stanford_media.settings');
+  foreach ($media_settings->get('embeddable_image_styles') as $image_style) {
+    $display_modes[] = _stanford_media_post_update_8202_image_display_mode($image_style);
+  }
+  $display_modes = array_unique($display_modes);
+  $media_settings->delete();
+
+  /** @var \Drupal\filter\FilterFormatInterface $filter */
+  foreach (FilterFormat::loadMultiple() as $filter) {
+    $filters = $filter->get('filters');
+    if (isset($filters['entity_embed'])) {
+      $filters['media_embed'] = [
+        'id' => 'media_embed',
+        'provider',
+        'media',
+        'status' => TRUE,
+        'weight' => $filters['entity_embed']['weight'],
+        'settings' => [
+          'default_view_mode' => 'full',
+          'allowed_view_modes' => array_combine($display_modes, $display_modes),
+        ],
+      ];
+      // Make sure if the filter strips html tags, we need to keep the
+      // drupal-media tag and its attributes.
+      if (isset($filters['filter_html'])) {
+        $filters['filter_html']['settings']['allowed_html'] .= ' <drupal-media data-entity-type data-entity-uuid data-align data-caption data-* alt>';
+      }
+      $filter->set('filters', $filters);
+      $filter->save();
+    }
+  }
+}
+
+/**
+ * Convert drupal-entity tags in wysiwyg's to drupal-media tags.
+ */
+function stanford_media_post_update_8202(&$sandbox) {
+  if (!isset($sandbox['count'])) {
+    $sandbox['entities'] = _stanford_media_post_update_8202_entity_list();
+    $sandbox['count'] = count($sandbox['entities']);
+  }
+
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $entity_ids = array_splice($sandbox['entities'], 0, 25);
```

Comment:
Thanks for breaking this up into a batch. This could get quite big. Do you know if post_update hooks have a timeout limit or pick back up if they exceed limits?

Reply:
I think they do. I believe it won't mark the update function finished until the `$sandbox` say its finished

Reply:
Yeah, checking in to this, all of these post_update hooks get logged to the `key_value` table `existing_updates` field.